### PR TITLE
Implement `assert_trait_sub_any`

### PR DESCRIPTION
**Pull request by Nadrieril**
It is constructed quite straightforwardly from `assert_impl_any` by
quantifying using a generic function. Sadly it can't actually reuse
`assert_impl_any` directly because of "using parameter from outer
function" error.

### CODE REVIEW

File: src/assert_trait.rs [hunk 1]
```diff
@@ -103,3 +103,100 @@ macro_rules! assert_trait_super_all {
...
+/// [`Eq`]:         https://doc.rust-lang.org/std/cmp/trait.Eq.html
+/// [`PartialEq`]:  https://doc.rust-lang.org/std/cmp/trait.PartialEq.html
+#[macro_export]
+macro_rules! assert_trait_sub_any {
+    ($sub:path: $($super:path),+ $(,)?) => {
+        const _: fn() = || {
+            use $crate::_core::marker::PhantomData;
+            use $crate::_core::ops::Deref;
+
+            fn forall<T: $sub>() {
```
Comment (nvzqz) [inline src/assert_trait.rs]: Small nit: please put an underscore between words.

Comment (Nadrieril) [reply src/assert_trait.rs]: You mean `for_all` ? I'm used to seeing `forall` as a single keyword, e.g. in LaTeX `\forall`, or in Dhall `forall (T: Type) -> ...`, so I don't see it as a mistake, but I have no strong opinion on which one to prefer

Comment (nvzqz) [reply src/assert_trait.rs]: I'm used to (and prefer) the Rust style of always separating words. I've only seen LaTeX's `\forall` before.

**Comment by Nadrieril** [comment]
Well, if you're adding up integers you could just as well `||` booleans together. And the wrapper trick allows for a limited form of emulating specialization. So this idea isn't too different

**Comment by nvzqz** [comment]
Ah good point. Unfortunately `||` doesn't work in `const` yet, but `|` does.

**Pull request merged**